bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryDirectDataWriterHelper.java

```diff
@@ -253,6 +259,13 @@ public long finalizeStream() throws IOException {
     if (this.protoRows.getSerializedRowsCount() != 0) {
       sendAppendRowsRequest();
     }
+    try {
+      ApiFutures.allAsList(appendRowsFutures).get();
+    } catch (InterruptedException | ExecutionException e) {
+      throw new BigQueryConnectorException("Could not retrieve AppendRowsResponse", e);
+    } finally {
+      appendRowsExecutor.shutdown();
```

Comment:
should we add a check at other public methods to assert that no operation is done after the executor is shutdown?

Reply:
The executor will throw an exception if it is used after being shutdown.